### PR TITLE
feat(NODE-5272): do not create or drop ecc collections

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

The driver no longer creates an ecc collection when creating an encrypted collection.

Additionally, the QE v2 protocol requires 7.0+ servers.  This means that driver 4.x would be incompatible with QE unless we bump the max wire version for 4.x to 21.

Backport of [#3660](https://github.com/mongodb/node-mongodb-native/pull/3660)

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/operations/create_collection.ts [hunk 1]
```diff
@@ -126,12 +135,17 @@ export class CreateCollectionOperation extends CommandOperation<Collection> {
         db.s.client.options.autoEncryption?.encryptedFieldsMap?.[`${db.databaseName}.${name}`];
 
       if (encryptedFields) {
+        // Creating a QE collection required min server of 7.0.0
+        if (server.description.maxWireVersion < MIN_SUPPORTED_QE_WIRE_VERSION) {
```
Comment (nbbeeken) [inline src/operations/create_collection.ts:139]: Daria had a question about whether this was necessary

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: src/operations/create_collection.ts (hunk 1 above)
Comment (nbbeeken) [reply src/operations/create_collection.ts:139]: @baileympearson bump

**Review by nbbeeken — APPROVED** [review]

**Comment by nbbeeken** [comment]
@baileympearson Is "fix" correct here? asking because the original PR was a feat

**Comment by baileympearson** [comment]
I think so - why was the original PR a feat?  

- QE is experimental, so the breakage is okay
- this fixes a bug with QE and 7.0 servers, where we'd create extra unnecessary collections

**Pull request merged**